https://github.com/Themaister/Granite/pull/129

From 9e78ed6c4ec494d29a9341d127cf3131a2e22722 Mon Sep 17 00:00:00 2001
From: orbea <orbea@riseup.net>
Date: Mon, 22 Jan 2024 08:03:46 -0800
Subject: [PATCH] util: fix the musl build

samu: job failed with status 1: /usr/lib/ccache/bin/c++  -I/tmp/Granite/util -I/tmp/Granite/application/global -std=gnu++14 -Wshadow -Wall -Wextra -Wno-comment -Wno-missing-field-initializers -Wno-empty-body -fno-omit-frame-pointer -msse3 -ffast-math -MD -MT util/CMakeFiles/granite-util.dir/aligned_alloc.cpp.o -MF util/CMakeFiles/granite-util.dir/aligned_alloc.cpp.o.d -o util/CMakeFiles/granite-util.dir/aligned_alloc.cpp.o -c /tmp/Granite/util/aligned_alloc.cpp
/tmp/Granite/util/aligned_alloc.cpp: In function 'void* Util::memalign_alloc(size_t, size_t)':
/tmp/Granite/util/aligned_alloc.cpp:47:5: error: 'uintptr_t' was not declared in this scope
   47 |     uintptr_t addr = 0;
      |     ^~~~~~~~~
/tmp/Granite/util/aligned_alloc.cpp:27:1: note: 'uintptr_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   26 | #include <string.h>
  +++ |+#include <cstdint>
   27 | #ifdef _WIN32
/tmp/Granite/util/aligned_alloc.cpp:53:5: error: 'addr' was not declared in this scope
   53 |     addr = ((uintptr_t)ptr + sizeof(uintptr_t) + boundary) & ~(boundary - 1);
      |     ^~~~
/tmp/Granite/util/aligned_alloc.cpp:53:24: error: expected ')' before 'ptr'
   53 |     addr = ((uintptr_t)ptr + sizeof(uintptr_t) + boundary) & ~(boundary - 1);
      |            ~           ^~~
      |                        )
samu: subcommand failed
---
 util/aligned_alloc.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/util/aligned_alloc.cpp b/util/aligned_alloc.cpp
index 674aeb3c0..9f5900eca 100644
--- a/Source/3rdParty/mupen64plus-video-parallel/util/aligned_alloc.cpp
+++ b/Source/3rdParty/mupen64plus-video-parallel/util/aligned_alloc.cpp
@@ -21,6 +21,7 @@
  */

 #include "aligned_alloc.hpp"
+#include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
 #ifdef _WIN32
